go-chaos/internal/pods.go

```diff
@@ -121,32 +121,33 @@ func (c K8Client) RestartPodWithGracePeriod(podName string, gracePeriodSec *int6
 }
 
 func (c K8Client) AwaitReadiness() error {
-	retries := 0
-	maxRetries := 300 // 5 * 60s
-	for {
-		if retries >= maxRetries {
-			return errors.New("Awaited readiness of pods in namespace %s, but timed out after 30s")
-		}
-		if retries > 0 {
-			time.Sleep(1 * time.Second)
-		}
+	return c.AwaitReadinessWithTimeout(5 * time.Minute)
+}
 
-		brokersAreRunning, err := c.checkIfBrokersAreRunning()
-		if err != nil {
-			return err
-		}
+func (c K8Client) AwaitReadinessWithTimeout(timeout time.Duration) error {
+	timedOut := time.After(timeout)
+	ticker := time.Tick(1 * time.Second)
 
-		gatewaysAreRunning, err := c.checkIfGatewaysAreRunning()
-		if err != nil {
-			return err
-		}
+	// Keep checking until we're timed out
+	for {
+		select {
+		case <-timedOut:
+			return errors.New(fmt.Sprintf("Awaited readiness of pods in namespace %v, but timed out after %v", c.GetCurrentNamespace(), timeout))
+		case <-ticker:
```

Comment:
:+1: Yeah I thought also last week that we can write this better with channels thanks for improving this :rocket: